Sources/ArcGISToolkit/Components/OfflineMapAreasView/PreplannedMapModel.swift

```diff
@@ -111,18 +287,41 @@ extension PreplannedMapModel {
     }
 }
 
+private extension PreplannedMapModel {
+    enum PathComponents: String {
+        case package = "package"
+        case mmpk = "mmpk"
+    }
+}
+
+extension PreplannedMapModel: Hashable {
+    nonisolated public static func == (lhs: PreplannedMapModel, rhs: PreplannedMapModel) -> Bool {
+        lhs === rhs
+    }
+    
+    public func hash(into hasher: inout Hasher) {
+        hasher.combine(ObjectIdentifier(self))
+    }
+}
+
 /// A type that acts as a preplanned map area.
 protocol PreplannedMapAreaProtocol {
     func retryLoad() async throws
     
+    var mapArea: PreplannedMapArea? { get }
```

Comment:
This is subverting all the good work that the protocol is doing for us. This should be a red flag that something is wrong.

Remove this property and where it needs to be used, we need to rethink how to get that work done. For example maybe this function should go away:

```swift
    /// Creates the parameters to download a preplanned offline map.
    /// - Returns: The parameters to download a preplanned offline map.
    private func createParameters() async -> DownloadPreplannedOfflineMapParameters? {
        guard let preplannedMapArea = preplannedMapArea.mapArea,
              let offlineMapTask else { return nil }
        do {
            // Create the parameters for the download preplanned offline map job.
            let parameters = try await offlineMapTask.makeDefaultDownloadPreplannedOfflineMapParameters(
                preplannedMapArea: preplannedMapArea
            )
            // Set the update mode to no updates as the offline map is display-only.
            parameters.updateMode = .noUpdates
            
            return parameters
        } catch {
            // If creating the parameters fails, set the failure.
            self.result = .failure(error)
            return nil
        }
    }
```

And the `PreplannedMapAreaProtocol` should have a makeParameters() function. Or maybe in the function above you try to cast the protocol to a `PreplannedMapArea`.

Reply:
Good point, the `mapArea` property was removed and the protocol now has a `makeParameters()` method.